xds/internal/server/rds_handler.go

```diff
@@ -0,0 +1,137 @@
+/*
+ *
+ * Copyright 2021 gRPC authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *
+ */
+
+package server
+
+import (
+	"sync"
+
+	"google.golang.org/grpc/xds/internal/xdsclient"
+)
+
+// rdsHandlerUpdate wraps the full RouteConfigUpdate that are dynamically queried for a given
+// server side listener.
+type rdsHandlerUpdate struct {
+	updates map[string]xdsclient.RouteConfigUpdate
+	err     error
+}
+
+// rdsHandler handles any RDS queries that need to be started for a given server
+// side listeners Filter Chains (i.e. not inline).
+type rdsHandler struct {
+	parent *listenerWrapper
+
+	mu sync.Mutex
+
```

Comment:
Nix the newline separating the mutex from the fields that it protects. Grouping the mutex and the fields protected it  is common practice to avoid adding a comment saying "mu protects access to foo and bar".